state_synchronizer/src/coordinator.rs

```diff
@@ -391,6 +397,10 @@ impl<T: ExecutorProxyTrait> SyncCoordinator<T> {
             // if coordinator didn't make progress by expected time, issue new request
             if let Some(tst) = expected_next_sync {
                 if SystemTime::now().duration_since(tst).is_ok() {
+                    if let Some(peer_id) = self.last_requested_peer_id {
```

Comment:
there are some nasty cases where we'll penalize incorrect peer. Imagine you requested chunk from A, it sends you corrupted chunk, then we immediately ask for chunk from B. B has nothing new; then progress check comes in, will check that there's no progress and will penalize B instead of A